public/app/plugins/panel/text/models.gen.ts

```diff
@@ -0,0 +1,24 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+// NOTE: This file will be auto generated from models.cue
+// It is currenty hand written but will serve as the target for cuetsy
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+export const modelVersion = Object.freeze([1, 0]);
+
+export enum TextMode {
+  HTML = 'html',
+  Markdown = 'markdown',
+}
+
+export interface PanelOptions {
+  mode: TextMode;
+  content: string;
+}
```

Comment:
My above-recommended change would change this to:

```ts
export interface PanelOptions {
  mode: "html" | "markdown";
  content: string;
} 
```


I don't actually remember if typescript will consider values from the `TextMode` enum to be to be structurally equivalent, and thus assignable here, which is obviously the goal. If not...mmm, yeah, need to think a bit about that one.

Reply:
this works -- but would like to use this as a simple example of an explicit type enum.   (in typescript what you wrote and an enum defined with strings are actually identical)

Reply:
Mmmm...interesting. Yeah, we'll have to add support for this in cuetsy - effectively tell it to pop out a type with a particular name into the top-level scope, then reference it.